src/windows/views/webview.py

```diff
@@ -247,11 +247,13 @@ def update_clip_data(self, clip_json, only_basic_props=True, ignore_reader=False
         except Exception:
             # Failed to parse json, do nothing
             log.warning('Failed to parse clip JSON data', exc_info=1)
+            return
 
         # Search for matching clip in project data (if any)
-        existing_clip = Clip.get(id=clip_data["id"])
+        existing_clip = Clip.get(id=clip_data.get("id"))
```

Comment:
"id" does not always exist, for example when dropping a new file into the timeline